util/encoding.go

```diff
@@ -31,23 +30,18 @@ func (bs *BitStream) enlarge(n uint16) {
 		return
 	}
 
-	const expThreshold = 64
-	const step = 16
-	var temp []byte
-
-	if final <= expThreshold {
-		finalF := float32(final - 1)
-		// Find out the leftmost one's position.
-		pos := *(*int)(unsafe.Pointer(&finalF))>>23&255 - 127
-		size := 1
-		if pos >= 0 {
-			size = 1 << (pos + 1)
-		}
-		temp = make([]byte, final, size)
+	newCap := cap(bs.b)
+	doubleCap := newCap + newCap
+	if final > doubleCap {
+		newCap = final
 	} else {
-		size := (final-expThreshold+step-1)/step*step + expThreshold
-		temp = make([]byte, final, size)
+		if newCap < 32 {
+			newCap = doubleCap
+		} else {
+			newCap += newCap / 4
```

Comment:
This requires some work.

For example, if final = 70 and newCap = 40, then `final > doubleCap` is false, `newCap < 32` is false, so we have `newCap += newCap / 4` which makes  the new newCap = 50. The `make` will then fail as you try to make an array sized larger than its capacity.